### ISSUE TITLE
Playlist content corruption

### ISSUE THREAD

**Issue by capitaineflam25** [libretro/RetroArch#7543]
(Retroarch for Nintendo Switch 1.7.5, Switch firmware 6.0.1)

I reproduced this issue many time, i'm not sure how it happens (maybe the playlist is updated at the same time the cfg file is written ?)

Here is how the corrupted lbl file looks like, on the right:
(On the left is my playlist before it gets corrupted)

![image](https://user-images.githubusercontent.com/985702/48067922-c5569380-e1d1-11e8-919a-304aa300112c.png)

**Comment by twinaphex** [comment]
Whoops, that is not good. Looks like entries from the retroarch config file seep into the playlist file.

@bparker06, any thoughts on this?

**Comment by bparker06** [comment]
is this still a problem?

**Comment by capitaineflam25** [comment]
> 
> 
> is this still a problem?

my playlist are stable now, so i don't edit them anymore.
i may do some test to try reproduce that issue.

**Comment by treyturner** [comment]
This is still happening in 1.7.6 (2019-02-21) on Switch, and it happens during normal use after all playlist content has been imported, even without any crashes of any kind. I'll check my console lists, and all of a sudden some of them stop at a certain letter. When I inspect the .lpl in a text editor, I see there's some random text from another config file invalidating the JSON. This happens regularly, at least once every day or two. I've started to keep a backup of my playlists in another directory and replacing the corrupt ones when it happens.

**Comment by m4xw** [comment]
@treyturner You sure you dont just mix old and new builds?

**Comment by treyturner** [comment]
I'm not sure what that means. I have a build with that date for which I stuck the .nro in /switch and am running via title override. When I updated to 1.7.6 I cleared out my /retroarch folder and replaced it with the bundled one, then recopied all my /retroarch/cores/system files etc.

Here's an example of my Gameboy playlist being corrupted mid-stream by my Gameboy Color playlist.

```
    {
      "path": "/retroarch_config/roms/Game Boy/Super Hunchback (USA).zip#Super Hunchback (USA).gb",
      "label": "Super Hunchback (USA)",
      "core_path": "DETECT",
      "core_name": "DETECT",
      "crc32": "1A45706E|crc",
      "db_name": "Nintendo - Game Boy.lpl"
    },
    {
      "path": "/retroarch_config/roms/Game Boy/Super Chase H.Q. (USA, Europe).zip#Super Chase H.Q. (USA, Europe).gb",
      "label": "Super Chase H.Q. (USA, Europe)",
      "co{
  "version": "1.0",
  "items": [
    {
      "path": "/retroarch_config/roms/Game Boy Color/Mr. Driller (USA).zip#Mr. Driller (USA).gbc",
      "label": "Mr. Driller (USA)",
      "core_path": "DETECT",
      "core_name": "DETECT",
      "crc32": "492C0EBF|crc",
      "db_name": "Nintendo - Game Boy Color.lpl"
    }
  ]
}
 "DETECT",
      "crc32": "6B497842|crc",
      "db_name": "Nintendo - Game Boy.lpl"
    },
    {
      "path": "/retroarch_config/roms/Game Boy/Super Battletank - War in the Gulf (USA).zip#Super Battletank - War in the Gulf (USA).gb",
      "label": "Super Battletank (USA)",
      "core_path": "DETECT",
      "core_name": "DETECT",
      "crc32": "DA27E798|crc",
      "db_name": "Nintendo - Game Boy.lpl"
    },
```

**Comment by m4xw** [comment]
You have to ensure the /switch loader matches the nightlies.
exfat?

**Comment by treyturner** [comment]
Yes, exFAT.

**Comment by m4xw** [comment]
Theres your problem.

**Comment by 2xtime** [comment]
Hi, there are so many issues about playlists, i'm not sure where to post. I created a playlist using 'Playlist Buddy' (old format). I played a game then went back twice to the gamelist and a huge amount of the games were missing (I could only see letters A-I). I'm using the latest nightly build (28-2-19) on Android. This has also happened on Windows. I have seen a few similar reports about this. Would it have anything to do with the amount of entries in the playlist ? Mine has 3000! (ZX Spectrum).

EDIT: Since updating Retroarch, all my MAME artwork is not showing in the playlist? Do I have to start all this from scratch since you updated the playlist format ?
